pkg/models/repository_test.go

```diff
@@ -34,4 +35,43 @@ func (s *RepositorySuite) TestRepositoriesCreate() {
 	err := tx.Where("url = ?", repo.URL).First(&found).Error
 	assert.Nil(s.T(), err)
 	assert.NotEmpty(s.T(), found.UUID)
+	assert.True(s.T(), strings.HasSuffix(found.URL, "/")) // test trailing slash added during creation
+}
+
+func (s *ModelsSuite) TestCleanupURL() {
+	tx := s.tx
+	var found Repository
+
+	type TestCase struct {
+		given    string
+		expected string
+	}
+
+	testCases := []TestCase{
+		{
+			given:    "https://example1.com",
+			expected: "https://example1.com/",
+		},
+		{
+			given:    "   https://example2.com   ",
```

Comment:
Can you use a special use domain name here?
https://www.iana.org/assignments/special-use-domain-names/special-use-domain-names.xhtml